provisioning/resources/control-plane/snowplow-mini-control-plane-api.go

```diff
@@ -20,34 +20,271 @@
+  if (req.Method == "PUT") {
+    _, err := callRestartSPServicesScript()
+    if err != nil {
+      http.Error(resp, err.Error(), 400)
+      return
+    } else {
+      resp.WriteHeader(http.StatusOK)
+      io.WriteString(resp, "OK")
+    }
+  }
+}
+
+func uploadEnrichments(resp http.ResponseWriter, req *http.Request) {
+  if req.Method == "POST" {
+    req.ParseMultipartForm(32 << 20)
+    file, handler, err := req.FormFile("enrichmentjson")
+    if err != nil {
+      http.Error(resp, err.Error(), 400)
+      return
+    }
+    defer file.Close()
+    f, err := os.OpenFile(enrichmentsPath + "/" + handler.Filename, os.O_WRONLY|os.O_CREATE, 0666)
+    if err != nil {
+      http.Error(resp, err.Error(), 400)
+      return
+    }
+    defer f.Close()
+    fileContentBytes, err := ioutil.ReadAll(file)
+    fileContent := string(fileContentBytes)
+
+    if !isJSON(fileContent) {
+      http.Error(resp, "JSON is not valid", 400)
+      return      
+    }  
+
+    io.WriteString(f, fileContent)
+    //restart SP services to get action the enrichments
+    _, err = callRestartSPServicesScript()
+    resp.WriteHeader(http.StatusOK)
+    if err != nil {
+      http.Error(resp, err.Error(), 400)
+      return
+    }
+    resp.WriteHeader(http.StatusOK)
+    io.WriteString(resp, "uploaded successfully")
+    return
+  }
+}
+
+func addExternalIgluServer(resp http.ResponseWriter, req *http.Request) {
+  if req.Method == "POST" {
+    req.ParseForm()
+    igluServerUri := req.Form["iglu_server_uri"][0]
+    igluServerApikey := req.Form["iglu_server_apikey"][0]
+
+    shellScriptCommand := []string{scriptsPath + "/" +  "add_external_iglu_server.sh",
+                                    igluServerUri,
+                                    igluServerApikey,
+                                    configPath,
+                                    scriptsPath}
+    cmd := exec.Command("/bin/bash", shellScriptCommand...)
+    err := cmd.Run()
+    if err != nil {
+      http.Error(resp, err.Error(), 400)
+      return
+    }
+    //restart SP services to get action the external iglu server 
+    _, err = callRestartSPServicesScript()
+    resp.WriteHeader(http.StatusOK)
+    if err != nil {
+      http.Error(resp, err.Error(), 400)
+      return
+    }
+    resp.WriteHeader(http.StatusOK)
+    io.WriteString(resp, "added successfully")
+  }
+}
+
+func addIgluServerSuperUUID(resp http.ResponseWriter, req *http.Request) {
+  if req.Method == "POST" {
+    req.ParseForm()
+    igluServerSuperUUID := req.Form["iglu_server_super_uuid"][0]
+    shellScriptCommand := []string{scriptsPath + "/" +  "add_iglu_server_super_uuid.sh", 
+                                    igluServerSuperUUID, 
+                                    configPath}
+    cmd := exec.Command("/bin/bash", shellScriptCommand...)
+    err := cmd.Run()
+    if err != nil {
+      http.Error(resp, err.Error(), 400)
+      return
+    }
+    //restart SP services 
```

Comment:
This restart sp services block has been repeated a few times - lets move it to a private function so you can re-use it.

Reply:
I think it is better with this way because there is a function for calling restart services scripts and what I did in the these "restart SP services" blocks is that calling restartSPServices function and handling errors. Creating another function just for error handling block will be a little bit verbose I think.